capsules/extra/src/hmac.rs

```diff
@@ -146,10 +146,7 @@ impl<
                                 }
                             })
                         })
-                        .unwrap_or(Err(ErrorCode::RESERVE));
-                    if ret.is_err() {
-                        return ret;
-                    }
+                        .unwrap_or(Err(ErrorCode::RESERVE))?;
```

Comment:
Two better options

```suggestion
                        .or(Err(ErrorCode::RESERVE))?;
```

```suggestion
                        .map_err(|e| ErrorCode::RESERVE)?;
```

Reply:
@alevy The first one changes the semantics (and I'm not sure would pass the type-checker). For a `Result<T, E>`, `unwrap_or()` takes a `T` and returns either the `Ok(T)` contained in the result, or the default value. `or()` will return a different `Result<T, F>` passed in, in case the original result is an `Err(_)`.

Second case is good, but we should rename `e` or `_` to avoid an unused closure argument warning.

Reply:
You're right, the first one should not have an enclosing `Err`